jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java

```diff
@@ -46,41 +46,94 @@ public void hasDependencyWithGroupIdWithNulls() {
     // Given
-    // @formatter:off
-    new Expectations() {{
-      project.getDependencies(); result = Arrays.asList(
-          Dependency.builder().groupId("io.dep").build(),
-          Dependency.builder().groupId("io.dep").artifactId("artifact").version("1.3.37").build(),
-          Dependency.builder().groupId("io.other").artifactId("artifact").version("1.3.37").build()
-        );
-    }};
-    // @formatter:on
+    JavaProject project = mock(JavaProject.class);
+    when(project.getDependencies()).thenReturn( //
+        Arrays.asList( //
+            Dependency.builder().groupId("io.dep").build(), //
+            Dependency.builder().groupId("io.dep").artifactId("artifact").version("1.3.37").build(), //
+            Dependency.builder().groupId("io.other").artifactId("artifact").version("1.3.37").build() //
+        ));
     // When
     final boolean result = JKubeProjectUtil.hasDependencyWithGroupId(project, "io.dep");
     // Then
     assertThat(result).isTrue();
   }
 
   @Test
-  public void hasDependencyWithGroupIdWithNoDependency(@Mocked JavaProject project) {
+  public void hasDependencyWithGroupIdWithNoDependency() {
     // Given
-    // @formatter:off
-    new Expectations() {{
-      project.getDependencies(); result = Arrays.asList(
-          Dependency.builder().groupId("io.dep").build(),
-          Dependency.builder().groupId("io.dep").artifactId("artifact").version("1.3.37").build(),
-          Dependency.builder().groupId("io.other").artifactId("artifact").version("1.3.37").build()
-      );
-    }};
-    // @formatter:on
+    JavaProject project = mock(JavaProject.class);
+    when(project.getDependencies()).thenReturn( //
+        Arrays.asList( //
+            Dependency.builder().groupId("io.dep").build(), //
+            Dependency.builder().groupId("io.dep").artifactId("artifact").version("1.3.37").build(), //
+            Dependency.builder().groupId("io.other").artifactId("artifact").version("1.3.37").build() //
+        ));
     // When
     final boolean result = JKubeProjectUtil.hasDependencyWithGroupId(project, "io.nothere");
     // Then
     assertThat(result).isFalse();
   }
 
+  @Test
+  public void hasDependencyWithGroupIdMatching_withJavaEEDependency_shouldMatchJavaeeGroupIDRegexp() {
+    // Given
+    JavaProject project = mock(JavaProject.class);
+    when(project.getDependencies()).thenReturn( //
+        Arrays.asList( //
+            Dependency.builder().groupId("io.dep").build(), //
+            Dependency.builder().groupId("javax.servlet").artifactId("javax.servlet-api").version("3.1.0").build(), //
+            Dependency.builder().groupId("io.other").artifactId("artifact").version("1.3.37").build() //
+        ));
+    // When
+    final boolean result = JKubeProjectUtil.hasDependencyWithGroupIdMatching(project, JAVAEE_GROUPID_REGEXP);
+    // Then
+    assertThat(result).isTrue();
+  }
+
+  static final String JAVAEE_GROUPID_REGEXP = "javax([/\\.](activation" + //
```

Comment:
nit, could this be moved to top. Cpuld this be private?

Reply:
I made this private
But I feel like it should be kept here just after the first reference to it: to avoid readers to scroll up and down.